File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.Set;
import java.util.HashSet;


class Solution {
    public int solution(int[] nums) {
        Set<Integer> hSet = new HashSet<>();
        for (int i : nums)
            hSet.add(i);
        int cnt = nums.length / 2;
        return hSet.size() < cnt ? hSet.size() : cnt;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> us;
    int select = nums.size();
    for (int i = 0; i < select; i++)
        us.insert(nums[i]);
    int unique = us.size();
    select /= 2;
    return select < unique ? select : unique;
}